packages/manager-components/src/hooks/iam/useOvhIam.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  useInfiniteQuery,
+  useQuery,
+  keepPreviousData,
+} from '@tanstack/react-query';
+import { IamInterface, PageInterface, PageReduceIam } from './iam.interface';
+
+export const fetchAuthorizationsCheck = async ({
+  actions,
+  urns,
+}: {
+  actions: string[];
+  urns: string[];
+}) => {
+  const { data } = await apiClient.v2.post(`/iam/authorization/check`, {
+    actions,
+    resourceURNs: urns,
+  });
+  return data;
+};
+
+function reducePages(pages: PageInterface[]) {
+  const mergedPages = pages.reduce(
+    (acc: PageReduceIam, page: PageInterface) => {
+      if (acc[page.resourceURN]) {
+        acc[page.resourceURN].authorizedActions = [
+          ...new Set([
+            ...acc[page.resourceURN].authorizedActions,
+            ...page.authorizedActions,
+          ]),
+        ];
+        acc[page.resourceURN].unauthorizedActions = [
+          ...new Set([
+            ...acc[page.resourceURN].unauthorizedActions,
+            ...page.unauthorizedActions,
+          ]),
+        ];
+      } else {
+        acc[page.resourceURN] = {
+          ...page,
+          authorizedActions: [...new Set(page.authorizedActions)],
+          unauthorizedActions: [...new Set(page.unauthorizedActions)],
+        };
+      }
+      return acc;
+    },
+    {},
+  );
+  const uniquePages = Object.values(mergedPages);
+  return uniquePages;
+}
+export function useInfiniteAuthorizationsIam({
+  actions = [],
+  urns = [],
+  isTrigger = true,
+  pageIndex = 0,
+}: IamInterface) {
+  const [authorizedActions, setAuthorizedActions] = useState([]);
+  const [previousPageIndices, setPreviousPageIndices] = useState([]);
+
+  const { data, fetchNextPage } = useInfiniteQuery({
+    queryKey: ['iam-authorization-infinite'],
+    queryFn: () => fetchAuthorizationsCheck({ actions, urns }),
+    initialPageParam: pageIndex,
+    enabled: urns.length > 0 && actions.length > 0 && isTrigger,
+    getNextPageParam: () => pageIndex,
+  });
+
+  useEffect(() => {
+    if (data) {
+      const pagesFlatten = data?.pages.map((page) => page).flat();
+      setAuthorizedActions(reducePages(pagesFlatten));
+      setPreviousPageIndices(data?.pageParams);
+    }
+  }, [data]);
+
+  return { authorizedActions, fetchNextPage, previousPageIndices };
+}
+
+export function useIsAuthorized(action: string, urn: string) {
+  const { authorizedActions } = useInfiniteAuthorizationsIam({});
+
+  let urnMapped: boolean[] = [];
+  if (authorizedActions && Object.values(authorizedActions).length > 0) {
+    urnMapped = authorizedActions?.map(
+      (element: PageInterface) =>
+        element.resourceURN === urn &&
+        element.authorizedActions.indexOf(action) > -1,
+    );
+  }
+  return urnMapped.indexOf(true) > -1;
+}
+
+export function useAuthorizationsIam({
+  actions,
+  urns,
+  isTrigger = true,
+}: IamInterface) {
+  const [authorizedActions, setAuthorizedActions] = useState([]);
+  const { data } = useQuery({
+    queryKey: ['iam-authorization', urns],
```

Comment:
**nitpick:** `actions` must be included in the same `queryKey` as well. Otherwise, when you make a request for different actions for same `resourceURNs`, data returned from the hook is wrong.